core/templates/dev/head/services/IdGenerationServiceSpec.js

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for IdGenerationService.
+ */
+
+describe('IdGenerationService', function() {
+  var IdGenerationService = null;
+
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    IdGenerationService = $injector.get('IdGenerationService');
+  }));
+
+  it('should test a random id generated', function() {
```

Comment:
`should generate a random id`., Also can we test this too:
id1 = genrate random id
id2 = generate random id
expect id1 to be not equal to id2.

Reply:
I think there is a very minor chance of two ids to be equal even if it is very low as it is generated randomly.. Should I still add this test? And regarding the length, I first thought to be of 11 as a string of length 11 was generated when I ran front end tests locally but the Travis failed as it generated id of length 10. You can see this in my previous commit. So I can't guarantee a fixed length.

Reply:
@lilithxxx, We are expecting this service/function to provide us with a unique Id. There is very (a very less 1/36^~~10~~11) less chance for the collision as of now! I'll suggest you to go adead and test id1 != id2.

Also, you can leave the length part (which I think is dynamic (between 10 and 11)).